lib/update-globals.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const fs = require('fs');
+const vm = require('vm');
+const path = require('path');
+const https = require('https');
+
+// collect globals
+const jsGlobals = {};
+
+vm.createContext(jsGlobals);
+vm.runInContext(`
+  this.props = Object.getOwnPropertyNames(this)`, jsGlobals);
+
+const windowGlobals = new Promise((resolve, reject) => {
+  https.get('https://tidoust.github.io/reffy-reports/whatwg/crawl.json',
+      (response) => {
+        let data = [];
+
+        response.on('data', (d) => {
+          data += d;
+        });
+
+        response.on('end', () => {
+          resolve(JSON.parse(data));
+        });
+      }).on('error', (err) => {
+    console.error('could not get WHATWG globals');
+    reject(err);
+  });
+});
+
+// create a global.json file (overwriting if it exists)
+/**
+ * Combines the globals into a stringified Set
+ *
+ * @param {Array} arr - an array of global variables
+ *
+ * @return {String} a stringified set of globals
+ */
+function listGlobals(arr) {
+  let str = `new Set([`;
+  arr.forEach((elt) => {
+    str += `'${elt}',\n    `;
+  });
```

Comment:
I think this might be simpler with `array.join()`, in particular, then you wouldn't need the `.substr()` at the end.